cdap-app-fabric/src/test/java/io/cdap/cdap/scheduler/CoreSchedulerServiceTest.java

```diff
@@ -596,7 +600,13 @@ private void testNewPartition(int expectedNumRuns) throws Exception {
   }
 
   private void waitForCompleteRuns(int numRuns, final ProgramId program) throws Exception {
-    Tasks.waitFor(numRuns, () ->  getRuns(program, ProgramRunStatus.COMPLETED), 30, TimeUnit.SECONDS);
+    try {
+      Tasks.waitFor(numRuns, () -> getRuns(program, ProgramRunStatus.COMPLETED), 30, TimeUnit.SECONDS);
+    } catch (Exception e) {
+      store.getRuns(program, ProgramRunStatus.ALL, 0, Long.MAX_VALUE, Integer.MAX_VALUE)
```

Comment:
let's add one generic log statement above this about something being wrong (in case there are 0 runs)

Reply:
done